Refactor WithRealmImport to accept file or directory

paulomorgado opened the pull request:
## Description

Updated the `WithRealmImport` method to change the parameter name from `importDirectory` to `import`, allowing it to accept either a directory or a single import file. Adjusted the method logic to handle both cases and updated the XML documentation for clarity.

Fixes #5086

/cc @davidfowl @DamianEdwards 

## Checklist

- Is this feature complete?
  - [ ] Yes. Ready to ship.
  - [x] No. Follow-up changes expected.
- Are you including unit tests for the changes and scenario tests if relevant?
  - [ ] Yes
  - [x] No
- Did you add public API?
  - [x] Yes
    - If yes, did you have an API Review for it?
      - [ ] Yes
      - [x] No
    - Did you add `<remarks />` and `<code />` elements on your triple slash comments?
      - [ ] Yes
      - [x] No
  - [ ] No
- Does the change make any security assumptions or guarantees?
  - [ ] Yes
    - If yes, have you done a threat model and had a security review?
      - [ ] Yes
      - [ ] No
  - [x] No
- Does the change require an update in our Aspire docs?
  - [x] Yes
    - Is this introducing a breaking change?
      - [ ] Yes
        - Link to aspire-docs issue (please use this [`breaking-change` template](https://github.com/dotnet/docs-aspire/issues/new?template=04-breaking-change.yml)):
      - [ ] No
        - Link to aspire-docs issue (please use this [`doc-idea` template](https://github.com/dotnet/docs-aspire/issues/new?template=02-docs-request.yml)):
  - [ ] No

paulomorgado committed:
Refactor WithRealmImport to accept file or directory

Updated the `WithRealmImport` method to change the parameter name from `importDirectory` to `import`, allowing it to accept either a directory or a single import file. Adjusted the method logic to handle both cases and updated the XML documentation for clarity.

paulomorgado committed:
Refactor Keycloak tests and update LUTConfig settings

Updated `WithRealmImportShouldThrowWhenBuilderIsNull` test
to check for null import instead of import directory, and
changed the expected exception type for non-existent imports.
Added a new configuration section in `Aspire.lutconfig`
to enable parallel builds, parallel test runs, and set a
test case timeout of 180 seconds.

DamianEdwards commented on Aspire.lutconfig:
```diff
@@ -0,0 +1,6 @@
+<LUTConfig Version="1.0">
```
Why was this added?

paulomorgado replied:
Love LUT, but still struggle with this new version.

DamianEdwards commented on tests/Aspire.Hosting.Keycloak.Tests/KeycloakPublicApiTests.cs:
```diff
@@ -107,26 +107,26 @@ public void WithRealmImportShouldThrowWhenBuilderIsNull()
 
     [Fact]
-    public void WithRealmImportShouldThrowWhenImportDirectoryDoesNotExist()
+    public void WithRealmImportShouldThrowWhenImportDoesNotExist()
```
Seems we have no tests that actually assert the happy path behavior! Could add a test/tests to verify that calling it with a directory or file that **does** exist adds the expected annotation?

paulomorgado replied:
Added!

paulomorgado committed:
Refactor Keycloak import tests and update LUTConfig

Removed obsolete configuration from Aspire.lutconfig.
Added new parameterized tests in KeycloakPublicApiTests.cs
for realm import functionality with directory and file inputs.
Removed the outdated WithRealmImportAddsBindMountAnnotation
test from KeycloakResourceBuilderTests.cs.

davidfowl commented on src/Aspire.Hosting.Keycloak/KeycloakResourceBuilderExtensions.cs:
```diff
@@ -146,20 +146,26 @@ public static IResourceBuilder<KeycloakResource> WithDataBindMount(this IResourc
     public static IResourceBuilder<KeycloakResource> WithRealmImport(
         this IResourceBuilder<KeycloakResource> builder,
-        string importDirectory,
+        string import,
```
This is a breaking change if you use named parameters but this package is preview so care level is low 😄

paulomorgado replied:
> This is a breaking change if you use named parameters but this package is preview so care level is low 😄

Yes and yes. But I couldn't keep `importDirectory` when the semantics has changed.

davidfowl commented:
/azp run
